### PR TITLE
gh-101979: fix a bug that parentheses in metavar argument of add_argument() were dropped

### PR DESCRIPTION
**Pull request by yeojin-dev**
Remove a regular expression that removes parentheses. Instead, add parentheses only if you use two or more non required and non suppressed actions.

* Issue: gh-101979

### CODE REVIEW

File: Lib/argparse.py [hunk 1]
```diff
@@ -405,8 +405,15 @@ def _format_actions_usage(self, actions, groups):
             else:
                 end = start + len(group._group_actions)
```
Comment (corona10) [inline Lib/argparse.py]: Please declare at this line
`group_action_count = len(group._group_actions)`

File: Lib/argparse.py [hunk 2]
```diff
@@ -405,8 +405,15 @@ def _format_actions_usage(self, actions, groups):
...
                 if actions[start:end] == group._group_actions:
+
+                    suppressed_actions_count = 0
                     for action in group._group_actions:
                         group_actions.add(action)
+                        if action.help is SUPPRESS:
+                            suppressed_actions_count += 1
+
+                    exposed_actions_count = len(group._group_actions) - suppressed_actions_count
```
Comment (corona10) [inline Lib/argparse.py]:
```suggestion
                    exposed_actions_count = group_action_count - suppressed_actions_count
```

File: Lib/argparse.py (hunk 1 above)
Comment (yeojin-dev) [reply Lib/argparse.py]: Good. I will apply it.

File: Misc/NEWS.d/next/Library/2023-02-28-09-52-25.gh-issue-101979.or3hXV.rst [hunk 3]
```diff
@@ -0,0 +1,2 @@
+Fix a bug that parentheses in metavar argument of :func:`add_argument()` were
```
Comment (JelleZijlstra) [inline Misc/NEWS.d/next/Library/2023-02-28-09-52-25.gh-issue-101979.or3hXV.rst]:
```suggestion
Fix a bug where parentheses in the ``metavar`` argument to :meth:`argparse.ArgumentParser.add_argument` were
```

File: Lib/test/test_argparse.py [hunk 4]
```diff
@@ -3764,6 +3764,28 @@ class TestHelpUsage(HelpTestCase):
     version = ''
 
 
+class TestHelpUsageWithParentheses(HelpTestCase):
```
Comment (JelleZijlstra) [inline Lib/test/test_argparse.py:3767]: I don't see any tests that actually add parentheses for a group with more than one action. Could you add one?

**Review by JelleZijlstra — COMMENTED** [review]
Looks good apart from minor

File: Lib/test/test_argparse.py (hunk 4 above)
Comment (yeojin-dev) [reply Lib/test/test_argparse.py:3767]: That's a good idea. But I think the testcase you want already exists using the code below. Please review it.

https://github.com/python/cpython/blob/66aa78cbe604a7c5731f074b869f92174a8e3b64/Lib/test/test_argparse.py#L2777

**Comment by yeojin-dev** [comment]
I have made the requested changes; please review again

File: Lib/test/test_argparse.py (hunk 4 above)
Comment (JelleZijlstra) [reply Lib/test/test_argparse.py:3767]: You're right, thanks! There are a lot of tests, I must have glanced over that one.

**Review by JelleZijlstra — APPROVED** [review]

**Pull request merged**